hw4/hw4.py

```diff
@@ -56,75 +43,59 @@ def lr(df):
     return
 
 
-if __name__ == "__main__":
-    """
-    diabetes = datasets.load_diabetes(as_frame=True)
-    df = diabetes['frame']
-    predictors = ['age','sex','bmi','bp','s1','s2','s3','s4','s5','s6']
-    response = ['target']
-    X = diabetes.data.values
-    y = diabetes.target.values
-    """
-    # iris = datasets.load_iris(as_frame=True)
-    # df = iris['frame']
-    # predictors = ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)','petal width (cm)']
-    # response = ['target']
-    # variable_type(df,predictors,response)
-    # lr(X,y)
-
-
-from sklearn.ensemble import RandomForestClassifier
+def random_forest():
+    import time
 
-# diabetes = datasets.load_diabetes(as_frame=True)
-# df = diabetes['frame']
-X = df.iloc[:, :-1]
-y = df.iloc[:, -1]
-from sklearn.model_selection import train_test_split
+    import numpy as np
+    from sklearn.ensemble import RandomForestClassifier
+    from sklearn.inspection import permutation_importance
+    from sklearn.model_selection import train_test_split
 
-X_train, X_test, y_train, y_test = train_test_split(
-    X, y, test_size=0.3, random_state=42
-)
-forest = RandomForestClassifier(random_state=0)
-forest.fit(X_train, y_train)
-import time
+    # diabetes = datasets.load_diabetes(as_frame=True)
+    # df = diabetes['frame']
+    X = df.iloc[:, :-1]
+    y = df.iloc[:, -1]
 
-import numpy as np
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=0.3, random_state=42
+    )
+    forest = RandomForestClassifier(random_state=0)
+    forest.fit(X_train, y_train)
+
+    start_time = time.time()
+    importances = forest.feature_importances_
+    std = np.std([tree.feature_importances_ for tree in forest.estimators_], axis=0)
+    elapsed_time = time.time() - start_time
```

Comment:
Not too sure why you are measuring the time for the classifier? Might be good if you are comparing different models together!